app/components/cityWeather.js

```diff
@@ -0,0 +1,74 @@
+'use client';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+	Sparklines,
+	SparklinesReferenceLine,
+	SparklinesLine,
+	SparklinesCurve,
+} from 'react-sparklines';
+
+export default function CityWeather() {
+	const cities = useSelector((state) => state.cities.cities);
+	//function to get averages of weather data
+	const getAverage = (array) =>
+		Math.round(
+			array.reduce((sum, currentValue) => sum + currentValue, 0) /
+				array.length
+		);
+	//displays cities weather with charts if they have been passed in. Has averages below each chart. If no info displays no weather data
+	const renderCities = () => {
+		if (cities.length > 0) {
+			return cities.map((city, index) => {
+				return (
+					<tbody key={index}>
+						<tr>
+							<th scope='row'>{city.cityName}</th>
+							<td>
+								<Sparklines data={city.temp}>
+									<SparklinesLine color='blue' />
+									<SparklinesReferenceLine type='avg' />
+								</Sparklines>
+							</td>
+							<td>
+								<Sparklines data={city.pressure}>
+									<SparklinesLine color='red' />
+									<SparklinesReferenceLine type='avg' />
+								</Sparklines>
+							</td>
+							<td>
+								<Sparklines data={city.humidity}>
+									<SparklinesLine color='yellow' />
+									<SparklinesReferenceLine type='avg' />
+								</Sparklines>
+							</td>
```

Comment:
DRY, don't repeat yourself, use JS helpers and components to make it more readable